widget/radio.go

```diff
@@ -279,19 +279,8 @@ func (r *Radio) CreateRenderer() fyne.WidgetRenderer {
 	return &radioRenderer{items, objects, r}
 }
 
-// SetMandatory sets if the radio must have a selection.
-func (r *Radio) SetMandatory(m bool) {
-	r.Mandatory = m
-	if r.Selected == "" && len(r.Options) > 0 {
-		r.Selected = r.Options[0]
-	}
-}
-
 // SetSelected sets the radio option, it can be used to set a default option.
 func (r *Radio) SetSelected(option string) {
-	if option == "" && r.Mandatory && len(r.Options) > 0 {
```

Comment:
I think removing this completely may have been too far - it now allows programattic setting of an invalid value ("").
Should it perhaps be re-instated without the len(r.Options) check to simply return, thus ensuring that a selected value cannot be unselected?

I guess I don't know for sure - but I wanted to ask the question in case this was an oversight

Reply:
(from slack conversation - yes, this is not an oversight)